Gems/Terrain/Code/Source/TerrainSystem/TerrainSystemBus.h

```diff
@@ -65,8 +66,16 @@ namespace Terrain
 
         virtual ~TerrainAreaHeightRequests() = default;
 
-        // Synchronous single input location.  The Vector3 input position versions are defined to ignore the input Z value.
```

Comment:
Is this no longer true? Think I'm relying on it ignoring the Z value for the ray cast stuff.

Reply:
This should be one API level down from what you're calling.  This is at the component level, not the terrain system level.  

That said, the "ignore Z" is no longer true because clearing it out is technically a bug, I think.  The underlying gradients that are being queried can be defined to be 2D or 3D gradients, so for 3D gradients if we're clearing out the input Z position we're going to force them to be 2D gradients.  That said, we might end up revisiting this a bit to fill in the input Z position with the component's Z position or something, because I'm not 100% sure it's right to pass down the input query Z position either, but for now I wanted to at least make it _not_ clear the Z to zero.

Reply:
I thought the original comment was referring to the inPosition z value, not the outPosition one?

The ray cast stuff should work regardless of whether it clears the output z or not, I just brought it up in case something has changed such that it no longer ignores the inPosition z value when calculating the height, but looking at the review I don't think that's the case so it should all be fine.

Reply:
This change won't affect the raycast - it's affecting the choice of underlying heights within the terrain system itself. The subtle bugfix I made was that previously, if the underlying gradient was a 3D gradient, moving the terrain spawner up and down would still get the same height values from the gradient.  Now, moving the terrain spawner up and down will get _different_ height values.  With a 2D gradient (the common case), everything works the same either way.

The raycast is querying a different API, which is asking the terrain _system_ for heights, not the terrain _spawner_.  The terrain _system_ queries each spawner based on the spawner's min bounding box height, regardless of what was passed in.  So the raycast can continue to assume the passed-in height doesn't matter.